derive/src/compile_bytecode.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{extract_spans, Diagnostic};
```

Comment:
Perhaps you could add here some documentation text describing the purpose and usage of this code. Something like how you described the macro usage in this pull request would be handy to keep as a reference in our source code as well.